ui/src/datasets.tsx

```diff
@@ -225,9 +227,48 @@ export function Datasets() {
           </Paper>
         </Grid>
         <Grid item xs={1}>
-          <Stack direction="row" alignItems="baseline">
-            <Typography variant="h4">3. Values</Typography>
-            <Typography variant="h5">(Columns)</Typography>
+          <Stack
+            direction="row"
+            alignItems="center"
+            justifyContent="space-between"
+          >
+            <Stack direction="row" alignItems="center">
+              <Typography variant="h4" mr={1}>
+                3. Values
+              </Typography>
+              <Typography variant="h5">(Columns)</Typography>
+            </Stack>
+            <Stack direction="row">
+              <Checkbox
+                size="small"
+                fontSize="inherit"
+                name="select-all-values"
+                checked={selectAllChecked}
+                onChange={() =>
+                  updateExcludedAttributes((selection) => {
+                    conceptSetEntities.map((entity) => {
+                      entity.attributes.map((attribute) => {
+                        if (!selection?.get(entity.name)) {
```

Comment:
Instead of toggling each attribute individually, I think you can simplify this by either clearing the selection or creating a new selection including all of the attributes. Immerjs allows you to return a new object instead of updating the passed in one which may help.

If that's not actually an improvement, now that this logic is being copied to a second place refactor it into a separate function inside this component. Something like `setAttributeSelected(entity: string, attribute: string, selected: boolean)`. Then use it both here and below.

Note that you don't need to check `attributes?.has(attribute)` before adding or deleting. Duplicates and non-existent elements will automatically be handled by the Set. It was only checked below since Set doesn't have a built in toggle method.

Reply:
I thought your first suggestion was a smart idea. I refactored the logic for the checkbox so that `excludedAttributes` is either cleared or filled with all attributes depending on the previous state of the "select all" checkbox.